lib/cdhc/enormp.c

```diff
@@ -49,7 +49,7 @@ double Cdhc_enormp(double x)
     }
     else {
         x2 = x * x;
-        x4 *= 1.;               /* huh, what? See original FORTRAN */
+        // x4 *= 1.;               /* huh, what? See original FORTRAN */ // 'x4' is uninitialized when used here
```

Comment:
This line does make little sense.
I found an original source in Internet Archive for [http://sunsite.univie.ac.at/statlib/general/cdh](https://web.archive.org/web/20051001000000*/http://sunsite.univie.ac.at/statlib/general/cdh)
which seem close enough, where the line is stated as `x4=1.0D0*x4`, for all code of the function:

```fortran
      FUNCTION enormp(x)
      REAL x
      DOUBLE PRECISION x1,x2,x3,x4,yy1,yy2
      xp1 = 0.771058495001320D-04
      xp2 = -0.00133733772997339D0
      xp3 = 0.0323076579225834D0
      xp4 = 0.0479137145607681D0
      xp5 = 0.128379167095513D0
      xq1 = 0.00301048631703895D0
      xq2 = 0.0538971687740286D0
      xq3 = 0.375795757275549D0
      xr1 = -1.36864857382717D-07
      xr2 = 0.564195517478974D0
      xr3 = 7.21175825088309D0
      xr4 = 43.1622272220567D0
      xr5 = 152.989285046940D0
      xr6 = 339.320816734344D0
      xr7 = 451.918953711873D0
      xr8 = 300.459261020162D0
      xs1 = 1.0D0
      xs2 = 12.7827273196294D0
      xs3 = 77.0001529352295D0
      xs4 = 277.585444743988D0
      xs5 = 638.980264465631D0
      xs6 = 931.354094850610D0
      xs7 = 790.950925327898D0
      xs8 = 300.459260956983D0
      xt1 = 2.10144126479064D0
      xt2 = 26.2370141675169D0
      xt3 = 21.3688200555087D0
      xt4 = 4.65807828718470D0
      xt5 = 0.282094791773523D0
      xu1 = 94.1537750555460D0
      xu2 = 187.114811799590D0
      xu3 = 99.0191814623914D0
      xu4 = 18.0124575948747D0
      x3  = 0.564189583547756D0
      x1 = abs(x)
      IF (x1.GT.0.5D0) GO TO 10
      x4 = x*x
      yy1 = ((((xp1*x4+xp2)*x4+xp3)*x4+xp4)*x4+xp5) +1.0D0
      yy2 = (((xq1*x4+xq2)*x4+xq3)*x4) + 1.0D0
      enormp = x* (yy1/yy2)
      RETURN
   10 IF (x1.GT.4.0D0) GO TO 20
      yy1 = ((((((xr1*x1+xr2)*x1+xr3)*x1+xr4)*x1+xr5)*x1
     +    + xr6)*x1+xr7)*x1+xr8
      yy2 = ((((((xs1*x1+xs2)*x1+xs3)*x1+xs4)*x1+xs5)*x1
     +    + xs6)*x1+xs7)*x1+xs8
      enormp =  1.0D0-exp(-x*x)*yy1/yy2
      IF (x .LT. 0.0D0) enormp = -enormp
      RETURN
   20 x2 = x*x
      x4=1.0D0*x4
      yy1 = ((((xt1*x4+xt2)*x4+xt3)*x4+xt4)*x4) + xt5
      yy2 = ((((xu1*x4+xu2)*x4+xu3)*x4+xu4)*x4) + 1.0D0
      enormp = (x3/x1)-(yy1*x1)/(x2*yy2)
      enormp = 1.0D0-exp(-x2)*enormp
      IF (x .LT. 0.0D0) enormp = -enormp
      RETURN
      END
```

Reply:
I'm curious on the origin of the following line `x4 = 0.5` (L53) which overrides the strange `x4 *= 1.`. Seems to predate version control, could track cdhd back to *GRASS 4.2.1 Version 10*.
@neteler Any idea?

Reply:
Oh, I added the file 23 years ago, in Dec 1999 :-D That was the first day of CVS being used (one day before year 2000 bug). See the log history...:

https://trac.osgeo.org/grass/log/grass/trunk/lib/cdhc/enormp.c

The initial version
- in SVN: https://trac.osgeo.org/grass/browser/grass/trunk/lib/cdhc/enormp.c?rev=9499
- in git: https://github.com/OSGeo/grass-legacy/commit/c0ac4dbb00f061734c83302a10a792208685b41a#diff-740f0803607b58d68b8eece29d6a9c077fa64f05d7ab7210693a75016dfbcd5c

Not too helpful, though. Here is the 4.3 changelog file (handwritten by me):
https://grass.osgeo.org/grass43/source/HISTORY.html
which contains:
`#87- added untested/sites/s.normal/cdhc-lib`

So this came from `s.normal` which included the cdhc library, written by Darrell McCauley.
Here some related email by him:
https://grass-user.osgeo.narkive.com/miLLR3i1/s-plus-and-grass#post2

with the link to his no longer existing web site (https://soils.ecn.purdue.edu/~mccauley/cdhc/).

Reply:
Thank you very much!